// ng-chartjs/esm2020/lib/ng-chartjs.module.mjs
import { NgModule } from '@angular/core';
import { NgChartjsDirective } from './ng-chartjs.directive';
import { NgChartjsService } from './ng-chartjs.service';
import { NgChartjsCustomPluginConfig } from './plugins-config';
import * as i0 from "@angular/core";
export class NgChartjsModule {
    /**
     * Register a plugin.
     * @param plugins
     */
    static registerPlugin(plugins = []) {
        const config = new NgChartjsCustomPluginConfig();
        config.plugins = plugins;
        return {
            ngModule: NgChartjsModule,
            providers: [
                {
                    provide: NgChartjsCustomPluginConfig,
                    useValue: config
                }
            ]
        };
    }
}
NgChartjsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: NgChartjsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgChartjsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.9", ngImport: i0, type: NgChartjsModule, declarations: [NgChartjsDirective], exports: [NgChartjsDirective] });
NgChartjsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: NgChartjsModule, providers: [NgChartjsService] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: NgChartjsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    declarations: [NgChartjsDirective],
                    exports: [NgChartjsDirective],
                    providers: [NgChartjsService]
                }]
        }] });
